ciao-image/datastore/ceph.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2016 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datastore
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path"
+
+	"github.com/01org/ciao/ciao-storage"
+)
+
+// Ceph implements the DataStore interface for Ceph RBD based storage
+type Ceph struct {
+	ImageTempDir string
+	BlockDriver  storage.CephDriver
+}
+
+// Write copies an image into the posix filesystem.
+// If the image already exists it will be overridden.
+func (c *Ceph) Write(ID string, body io.Reader) (int64, error) {
+	imagePath := path.Join(c.ImageTempDir, ID)
```

Comment:
You could use https://golang.org/pkg/io/ioutil/#TempFile.  Might save you one line of code.